packages/gatsby-transformer-cloudinary/gatsby-node.js

```diff
@@ -244,6 +248,40 @@ exports.onCreateNode = async ({
     });
   }
 };
+// The Old Way
+// onPreInit is setting 🌐 global state
+// so that other functions can access that global state. Which is The Old Way. And used here 😉
+
+let coreSupportsOnPluginInit = 'unstable' | 'stable' | undefined;
+
+try {
+  const { isGatsbyNodeLifeCycleSupported } = require(`gatsby-plugin-utils`);
+  if (isGatsbyNodeLifeCycleSupported(`onPluginInit`)) {
+    coreSupportsOnPluginInit = 'stable';
+  } else if (isGatsbyNodeLifeCycleSupported(`unstable_onPluginInit`)) {
+    coreSupportsOnPluginInit = 'unstable';
+  }
+} catch (error) {
+  console.error(
+    `Could not check if Gatsby supports onPluginInit lifecycle 🚴‍♀️  `,
+  );
+}
+
+let globalPluginOptions = {};
+
+const initializeGlobalState = (_, pluginOptions) => {
+  globalPluginOptions = pluginOptions;
```

Comment:
Move the code inside the current `onPreInit` below into this function and remove the current `onPreInit` below.